Formula/grafana-agent.rb

```diff
@@ -0,0 +1,98 @@
+class GrafanaAgent < Formula
+  desc "Exporter for Prometheus Metrics, Loki Logs, and Tempo Traces"
+  homepage "https://grafana.com/docs/agent/"
+  url "https://github.com/grafana/agent/archive/refs/tags/v0.21.2.tar.gz"
+  sha256 "a9deceaf1c09f19e5f44204e7d7c755dc24a5dd5a532c76c0524e762a695b590"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  on_linux do
+    depends_on "systemd" => :build
+  end
+
+  def install
+    ldflags = %W[
+      -X github.com/grafana/agent/pkg/build.Branch=HEAD
+      -X github.com/grafana/agent/pkg/build.Version=v#{version}
+      -X github.com/grafana/agent/pkg/build.Revision=v#{version}
```

Comment:
This seems more accurate
```suggestion
      -X github.com/grafana/agent/pkg/build.Revision=#{revision}
```

Reply:
@SMillerDev I was trying to approximate the output you'd get when building using their Makefile. Go applications typically use a `v` prefix. But I defer to you :)

Reply:
But revision doesn't seem like the same thing as "version", or are they just using it in an unconventional way here?

Reply:
@SMillerDev Oh apologies, I misread that part. You're right.

Reply:
Their build script is using the Git commit hash as the revision. `revision` in Homebrew is the package revision, I think. Should we just leave it blank or is there a way to get the commit hash?

Reply:
Not unless we switch to a git checkout, which seems like a lot of work for very little gain here. Let's leave it blank.

Reply:
Good call, pushed.